force-app/main/default/lwc/attendance/attendance.js

```diff
@@ -159,6 +183,25 @@ export default class Attendance extends LightningElement {
         );
     }
 
+    get hasPermissions() {
+        return this.isReadOnly
+            ? this.hasReadPermissions
+            : this.hasReadPermissions && this.hasWritePermissions;
+    }
+
+    get showUpdateButton() {
+        return this.isReadOnly && this.hasWritePermissions;
```

Comment:
I wonder if we should change the variable name from isReadOnly to isReadMode or something? Reading this line was a little confusing for me at first.